src/excalidraw-app/index.tsx

```diff
@@ -261,18 +265,22 @@ function ExcalidrawApp(props: { collab: CollabContext }) {
       onCollabButtonClick={collab.onCollabButtonClick}
       isCollaborating={collab.isCollaborating}
       onPointerUpdate={collab.onPointerUpdate}
+      {...props.testProps}
     />
   );
 }
 
 const AppWithCollab = (Component: typeof ExcalidrawApp) => {
-  return () => {
+  return <K extends keyof ExcalidrawProps>(props: {
+    /** for testing purposes */
+    testProps?: Pick<ExcalidrawProps, K>;
+  }) => {
```

Comment:
I couldn't figure out how in Jest/React-testing-lib we can pass props to nested components (I don't have much experience testing react compos), so I added `testProps`. If you have a better idea, that'd be great.

Reply:
I think when we are using `AppWithCollab`/ `ExcalidrawApp` we should not try to pass extra props to nesting comp (as that's not ideally supported and feels like hack) just for testing purpose so I instead mocked local storage for that case instead of passing extra props as that's the real world scenario as well.  And if we want to pass to the inner component I would rather prefer to use inner comp directly instead of passing props from nesting comp.
Let me know what you think